### ISSUE TITLE
How to Uninstall a Distro? / Required Folder

### ISSUE THREAD

**Issue by QNetSo** [MicrosoftDocs/WSL#664]
The directions do not point out that the install will then call the linux distro from where you extracted the files. Thus requiring that folder to remain for wsl to work. (%userprofile%/downloads/ubuntu) - you should specify to extract the files to a more suitable location. 

So that needs updated :) ... And how do I uninstall the distro so I can install it from a better folder location?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7bc0a885-286b-df36-acfd-f4afb5a33c9b
* Version Independent ID: 238333c7-116d-2697-702a-58fc7efe3ee9
* Content: [Install the Linux Subsystem on Windows Server](https://docs.microsoft.com/en-us/windows/wsl/install-on-server)
* Content Source: [WSL/install-on-server.md](https://github.com/MicrosoftDocs/WSL/blob/live/WSL/install-on-server.md)
* Product: **dev-environment**
* Technology: **windows-subsystem-for-linux**
* GitHub Login: @craigloewen-msft
* Microsoft Alias: **crloewen**

**Comment by QNetSo** [comment]
Well it appears that if I delete the key under here: HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Lxss that that uninstalls the distro.

**Comment by Biswa96** [comment]
Deleting the reigstry does not _uninstall_ (literally) the distribution in WSL. It will only _unregister_ it. The install path will not be deleted.

**Comment by johne53** [comment]
> There are two ways to uninstall a distro:
> 
> * If you downloaded it from the store, right click on the icon in your start menu (or search bar) and click 'Uninstall'.
> * You can also run `wsl --unregister <distroName>` to uninstall a distro which will remove those files for you.
> 

In my case I'm not seeing 'Uninstall' (in the 1st example) so I'd need the 2nd method - and I'm assuming that the line would need to get run form a Windows Command Prompt?

So I tried running it here to uninstall Ubuntu:-

- wsl --unregister ubuntu 

but it gives me this error:-  **There is no distribution with the supplied name**

I've tried upper case / lower case / capitalized etc but I see the same error each time. So how do I find the correct name to use? (presumably it's a bit more complicated than just **ubuntu** ??)

**Comment by eli2and40** [comment]
> 
> 
> > There are two ways to uninstall a distro:
> > 
> > * If you downloaded it from the store, right click on the icon in your start menu (or search bar) and click 'Uninstall'.
> > * You can also run `wsl --unregister <distroName>` to uninstall a distro which will remove those files for you.
> 
> In my case I'm not seeing 'Uninstall' (in the 1st example) so I'd need the 2nd method - and I'm assuming that the line would need to get run form a Windows Command Prompt?
> 
> So I tried running it here to uninstall Ubuntu:-
> 
>     * wsl --unregister ubuntu
> 
> 
> but it gives me this error:- **There is no distribution with the supplied name**
> 
> I've tried upper case / lower case / capitalized etc but I see the same error each time. So how do I find the correct name to use? (presumably it's a bit more complicated than just **ubuntu** ??)

What appears by the distro you want to uninstall when you enter "> wsl --list -v"? Perhaps your name should have version info?

**Comment by irmackenzie** [comment]
I tried uninstalling a distro using `wsl -u` command, but just got an error message "**user not found**", and the command started up the distro. I was able to uninstall via _Programs & Features_

**Comment by bosseks** [comment]
> 
> 
> I tried uninstalling a distro using `wsl -u` command, but just got an error message "**user not found**", and the command started up the distro. I was able to uninstall via _Programs & Features_

It looks like you are using the WSL1 version and therefore you have to write 'wslconfig /u <Distributionsname>'.
See 'wslconfig --help'

**Comment by marbaa** [comment]
Actually `wslconfig /u` or `wsl --unregister` will not physically delete files from disk. It will just disappear from `wsl --list` output. After running that command, I was still able to search files on disks and they were taking space.

So how can I properly delete distro? Which was installed manually by executing for example `kali-linux-08-06-2019.appx`?

**Comment by justinclift** [comment]
Yeah, the "unregister" command really doesn't do what the original poster asked.  The downloaded distribution is still physically on the disk and taking up space, not actually uninstalled.

Unregistering just seems to wipe the initial user config for the distribution.  Not real helpful towards removing the distribution from disk.  Which (so far), doesn't seem to have an non-extreme way to do (eg mount the disk in *nix and edit the file system directly).

**Comment by akochepasov** [comment]
This actually uninstalls the distro. Thanks

**Comment by marbaa** [comment]
After long time coming back asking again question from my previous post.. How can I physically delete distro from filesystem? Don't tell me that you invented something which can't be removed :).

**Comment by Biswa96** [comment]
1. Unregister the distribution with `wsl.exe --unregister <Distro>` command.
2. Uninstall the Microsoft Store package from Settings or Start Menu or `Remove-AppxPackage` Powershell command.

These steps are same since inception of WSL distributions in Microsoft Store.

**Comment by justinclift** [comment]
> Don't tell me that you invented something which can't be removed :).

**\*cough\*** Microsoft Edge **\*cough\*** :wink:

**Comment by Biswa96** [comment]
> *cough* Microsoft Edge *cough* 😉

`rm -rf "/mnt/c/Program Files (x86)/Microsoft"` 😎

**Comment by Anuj-02** [comment]
It's working perfectly, at least for me

**Comment by iamanshulgit** [comment]
It's working for me. I tried uninstalling it first, then I realized that the image has been removed but in wsl2 it's still showing. Then I ran the command : `wsl --unregister ubuntu`. It removed and cleared the source folder.

**Comment by bartpotrykus** [comment]
> It's working for me. I tried uninstalling it first, then I realized that the image has been removed but in wsl2 it's still showing. Then I ran the command : `wsl --unregister ubuntu`. It removed and cleared the source folder.

Agree. Done the same just a minute ago.
